### PR TITLE
Safename for scope variable shouldn't contain ".". Fixes #36377

### LINKED ISSUE #36377: [modeler] Any command with "." (e.g. grass') can't be used in expressions
**Describe the bug**
In modeler, if a grass algorithm is used, it's output can't be used in expression generating syntax expression error

**How to Reproduce**

1. Chosse a simple grass alg (e.g. r.cost)
2. create a model with inputs to feed the alg
3. add the grass alg and feed it with input from model
4. Add a new alg to the model having as input an expression getting ouput of previous grass alg (e.g. @r.cost_output) => expression syntax error due to "."

**QGIS and OS versions**

QGIS version | 3.13.0-Master | QGIS code revision | 1435076d2f
-- | -- | -- | --
Compiled against Qt | 5.9.5 | Running against Qt | 5.9.5
Compiled against GDAL/OGR | 2.2.3 | Running against GDAL/OGR | 2.2.3
Compiled against GEOS | 3.7.1-CAPI-1.11.1 | Running against GEOS | 3.7.1-CAPI-1.11.1 27a5e771
Compiled against SQLite | 3.22.0 | Running against SQLite | 3.22.0
PostgreSQL Client Version | 12.2 (Ubuntu 12.2-2.pgdg18.04+1) | SpatiaLite Version | 4.3.0a
QWT Version | 6.1.3 | QScintilla2 Version | 2.10.2
PROJ.4 Version | 493
OS Version | Ubuntu 18.04.4 LTS | This copy of QGIS writes debugging output.
Active python plugins | processing_whitebox; plugin_reloader; processing_taudem; quick_map_services; galiciasustentable; SpreadsheetLayers; firstaid; IPyConsole; qconsolidate; excel_sync; db_manager; processing; MetaSearch

### PR DESCRIPTION
**Pull request by luipir**
## Description

output variable names added in the algorithm sope are build from command name, and if the command is a grass command it contains "." generating error in expression parsing.
This PR add "." as sanitised character to the other chars already sanitised in the QgsProcessingModelAlgorithm::variablesForChildAlgorithm call

micro fix sponsored by GaliciaSustentable (https://fgsustentable.org/)

### CODE REVIEW

**Comment by m-kuhn** [comment]
does a `.` in a reg exp not need to be escaped?

**Comment by nyalldawson** [comment]
Needs a unit test too!

**Comment by luipir** [comment]
> Needs a unit test too!

I didn't added unit test becuase there weren't any unit test for that  function, not idea why!

**Comment by nyalldawson** [comment]
That's even more reason to add some!

**Comment by luipir** [comment]
> does a `.` in a reg exp not need to be escaped?

I can check escaping, but with my empiric test worked without escaping

**Comment by luipir** [comment]
@nyalldawson tests added
@m-kuhn . escaped

File: tests/src/analysis/testqgsprocessing.cpp [hunk 1]
```diff
@@ -9449,6 +9449,22 @@ void TestQgsProcessing::modelExecution()
   QGSCOMPARENEAR( variables.value( "SOURCE_LAYER_maxx" ).value.toDouble(), -83.3333, 0.001 );
   QGSCOMPARENEAR( variables.value( "SOURCE_LAYER_maxy" ).value.toDouble(), 46.8719, 0.001 );
 
+  // test safe name of the child alg parameter as source to another algoritmn
+  // parameter name should have [\s ' ( ) : .] chars changed to "_" (regexp [\\s'\"\\(\\):\.])
+  // this case is esecially important in case of grass algs where name algorithm contains "."
+  // name of the variable is get from childDescription or childId. To avoid using grass provider
+  // is simpler to use native algorithm and change it's description.
+  // test to check fix https://github.com/qgis/QGIS/issues/36377
```
Comment (nyalldawson) [inline tests/src/analysis/testqgsprocessing.cpp:9457]:
```suggestion
  // name of the variable is get from childDescription or childId. Refs https://github.com/qgis/QGIS/issues/36377
```

File: tests/src/analysis/testqgsprocessing.cpp [hunk 2]
```diff
@@ -9449,6 +9449,22 @@ void TestQgsProcessing::modelExecution()
...
+  QgsProcessingModelChildAlgorithm &cx1 = model2.childAlgorithm( "cx1" );
+  QString oldDescription = cx1.description();
+  cx1.setDescription( "cx '():.1" );
+  variables = model2.variablesForChildAlgorithm( "cx3", context );
+  QVERIFY( !variables.contains( "cx1_OUTPUT" ) );
+  QVERIFY( !variables.contains( "cx '():.1_OUTPUT" ) );
+  QVERIFY( variables.contains( "cx______1_OUTPUT" ) );
+  cx1.setDescription( oldDescription ); // set descrin back to avoid fail of following tests
+
+  // test to python convertion
```
Comment (nyalldawson) [inline tests/src/analysis/testqgsprocessing.cpp:9467]:
```suggestion
  // test model to python conversion
```

**Comment by luipir** [comment]
merged comments modifications

**Review by nyalldawson — APPROVED** [review]

**Comment by nyalldawson** [comment]
Unrelated test failure (pg raster)

**Comment by luipir** [comment]
Ho sorry @nyalldawson I'll squash myself next time, tnx

**Pull request merged**